### PR TITLE
Converting Date/Time to EST

### PR DESCRIPTION
**Pull request by ashley**
For Issue #170 
* Created functions in preProcess to reformat dates and times to EST.
* Implemented this function to created and modified fields for all forms (including default controller)

Question: preProcess.prettifyDate() can be replaced with moment functions. Should I refactored that?

### CODE REVIEW

File: app/index.html [hunk 1]
```diff
@@ -27,9 +27,14 @@
   <script src="/js/lib/chosen-angular.js"></script>
   <script src="//cdn.tinymce.com/4/tinymce.min.js"></script>
   <script src="/js/lib/angular-ui-tinymce.js"></script>
-
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/moment-timezone/0.5.11/moment-timezone.js"></script>
```
Comment (ashleytqy) [inline app/index.html]: is there a reason you're using a cdn for `moment-timezone`, but storing `moment` in `/js/lib`?

Comment (ashleytqy) [reply app/index.html]: also, would `https://github.com/urish/angular-moment` be more suitable?

File: app/js/controllers/actions/event/event-list-ctrl.js [hunk 2]
```diff
@@ -57,11 +58,22 @@ angular
 				}
 
 				_.each($scope.data, function(element) {
+					element = preProcess.changeDate(element);
+					$scope.eventDetails.time[element.id] = {};
+					$scope.eventDetails.time[element.id].start = preProcess.convertTimeToEST(element.attributes.startDateTime);
+					$scope.eventDetails.time[element.id].end = preProcess.convertTimeToEST(element.attributes.endDateTime);
 					//mapping eventID to venue names with links
 					if (element.relationships.venue.data !== null) {
 						// !! change venueURL to the venue page once the venue override page is ready
```
Comment (ashleytqy) [inline app/js/controllers/actions/event/event-list-ctrl.js]: you can get rid of this comment too ^ thanks!

File: app/js/controllers/actions/incomes/incomes-list-ctrl.js [hunk 3]
```diff
@@ -22,6 +22,8 @@ angular
 		}
 
 		_.each($scope.data, element => {
+			element = preProcess.changeDate(element);
+			element.attributes.recievedOn = preProcess.convertTimeToEST(element.attributes.recievedOn);
```
Comment (ashleytqy) [inline app/js/controllers/actions/incomes/incomes-list-ctrl.js]: why is this necessary / not covered in ` preProcess.changeDate(element)`?

File: app/js/services/preprocessing.js [hunk 4]
```diff
@@ -33,14 +33,38 @@ angular
...
+                return undefined;
+            }
+            var hour = parseInt(time.substring(11,13));
+            var minute = time.substring(14,16);
+            var night = 'AM';
+            if(hour > 12){
+                hour -= 12;
+                night = 'PM';
+            }
+            //return this.prettifyDate(time) + " " + hour + ":" + minute + night;
```
Comment (ashleytqy) [inline app/js/services/preprocessing.js]: comment!

File: app/js/services/preprocessing.js [hunk 5]
```diff
@@ -33,14 +33,38 @@ angular
...
+            return moment(time).tz('America/New_York').format('LLLL');;
```
Comment (ashleytqy) [inline app/js/services/preprocessing.js]: can we do `October 1, 2014 12:00 AM (Wednesday)` instead of `Wednesday, October 1, 2014 12:00 AM`?

File: app/js/controllers/actions/incomes/incomes-list-ctrl.js (hunk 3 above)
Comment (ashley) [reply app/js/controllers/actions/incomes/incomes-list-ctrl.js]: changeDate() is a helper function that called convertTimeToEST() except it only calls it on the element's create and modified attributes. Since some pages have other fields that require the conversion, you can use convertTimeToEST() on them in their controller.

File: app/js/services/preprocessing.js [hunk 6]
```diff
@@ -35,14 +35,37 @@ angular
...
         },
-				prettifyDate: function(date) {
-					if (date == undefined) { return; };
-					var monthNames = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
-					var year = parseInt(date.substring(0, 4));
-					var month = parseInt(date.substring(5, 7));
-					var day = parseInt(date.substring(8, 10));
-					return monthNames[month - 1] + " " + day + ", "+ year;
-				},
+        changeDate: function(element){
```
Comment (ashleytqy) [inline app/js/services/preprocessing.js]: maybe a better name would be `changeCreatedAndModifiedAttribute`. it's clearer / more specific than `changeDate` even though it's a bit long.

Comment (ashleytqy) [inline app/js/services/preprocessing.js]: maybe instead of `changeDate`, `changeCreatedAndModifiedAttributes` would be a more specific name even though it's longer

Comment (ashleytqy) [reply app/js/services/preprocessing.js]: what do you think of making this function more versatile as a wrapper function to `convertTimeToEST() `.

e.g. 
```js
convertTimeAttributes: function(element, ['created', 'modified', 'received']) {
  // basically what you have currently element.attributes['created'] = 
  this.convertTimeToEST(element.attributes['created'])
  // etc...
}
```

**Comment by ashleytqy** [comment]
> Question: preProcess.prettifyDate() can be replaced with moment functions. Should I refactored that?

Sure, go ahead!

File: app/js/controllers/actions/default/default-list-ctrl.js [hunk 7]
```diff
@@ -22,7 +22,7 @@ angular
 			$scope.model = _.find($scope.data, {id: resourceId});
 		}
 		_.each($scope.data, function(element) {
-			element = preProcess.changeDate(element);	
+			element = preProcess.convertTimeAtrtributes(element);
```
Comment (ashleytqy) [inline app/js/controllers/actions/default/default-list-ctrl.js]: spelling -> Attributes

File: app/js/services/preprocessing.js [hunk 8]
```diff
@@ -15,9 +15,9 @@ angular
             promise.then(function() {
                 Restangular.all(name)
                     .getList()
-                    .then(function(list){
-                        list.forEach(function(element) {
-                            objectIdToNameHash[element.id] = element.attributes.name;
+                    .then(function(element){
```
Comment (ashleytqy) [inline app/js/services/preprocessing.js]: can you keep the way it was before for line 18 to 20?

File: app/js/controllers/actions/incomes/incomes-list-ctrl.js [hunk 9]
```diff
@@ -22,6 +22,7 @@ angular
 		}
 
 		_.each($scope.data, element => {
+			element = preProcess.convertTimeAtrtributes(element, Array("recievedOn"));
```
Comment (ashleytqy) [inline app/js/controllers/actions/incomes/incomes-list-ctrl.js]: since you are using `...attributeArray`, "recievedOn" doesn't need to be in an array. u could do `element = preProcess.convertTimeAtrtributes(element, "recievedOn")`, unless you want to change the signature of the function

File: app/js/services/preprocessing.js [hunk 10]
```diff
@@ -35,14 +35,38 @@ angular
...
         },
-				prettifyDate: function(date) {
-					if (date == undefined) { return; };
-					var monthNames = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
-					var year = parseInt(date.substring(0, 4));
-					var month = parseInt(date.substring(5, 7));
-					var day = parseInt(date.substring(8, 10));
-					return monthNames[month - 1] + " " + day + ", "+ year;
-				},
+        convertTimeAtrtributes: function(element, ...attributeArray){
```
Comment (ashleytqy) [inline app/js/services/preprocessing.js]: rmb to update spelling for all of the places u r using `convertTimeAtrtributes` -> `convertTimeAttributes`
